### PR TITLE
Rename directories Solaris 11

### PR DESCRIPTION
**Pull request by DFolchA**
|Related issue|
|---|
|#570|

## Description
This PR adds the necessary changes to rename the `logs/ossec` and `queue/ossec `directories to Solaris 11 packages.

## Logs example

image

## Tests

- Build the package in any supported platform
  - [x] Solaris
- [x] Package installation
- [x] Package upgrade
- [x] Package downgrade
- [x] Package remove
- [x] Package install/remove/install

- Tests for Solaris
  - [x] Test the package on Solaris 10
  - [x] Test the package on Solaris 11
  - [x] Check file permissions on Solaris 11 template

### CODE REVIEW

**Review by VictorMorenoJimenez — APPROVED** [review]
:rocket:

File: solaris/solaris11/SPECS/template_agent_v4.2.0.json [hunk 1]
```diff
@@ -671,6 +671,14 @@
...
     },
+    "/var/ossec/logs/wazuh": {
+        "class": "static",
+        "group": "ossec",
+        "mode": "0750",
+        "prot": "drwxr-x---",
+        "type": "directory",
+        "user": "ossec"
+    },
     "/var/ossec/logs/ossec": {
```
Comment (okynos) [inline solaris/solaris11/SPECS/template_agent_v4.2.0.json:682]: Have we to remove `/var/ossec/logs/ossec`? and `/var/ossec` still being `ossec?`.

File: solaris/solaris11/generate_wazuh_packages.sh [hunk 2]
```diff
@@ -160,12 +163,24 @@ create_package() {
...
     mv wazuh-agent.p5m.1.aux.fixed wazuh-agent.p5m.1
-
     # Add the preserve=install-only tag to the configuration files
     for file in etc/ossec.conf etc/local_internal_options.conf etc/client.keys; do
         sed "s:file $file.*:& preserve=install-only:"  wazuh-agent.p5m.1 > wazuh-agent.p5m.1.aux_sed
         mv wazuh-agent.p5m.1.aux_sed wazuh-agent.p5m.1
     done
+
+    if [ "${major_version}" -ge "4" ]; then
+        if [ "${minor_version}" -ge "2" ]; then
```
Comment (okynos) [inline solaris/solaris11/generate_wazuh_packages.sh:173]: Where create we `/var/ossec/logs/wazuh` files?

File: solaris/solaris11/postinstall.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+# postinst script for wazuh-agent
+# Wazuh, Inc 2015-2020
+
+if [ -d <INSTALL_PATH>/logs/ossec ]; then
+  mv <INSTALL_PATH>/logs/ossec/* <INSTALL_PATH>/logs/wazuh
```
Comment (okynos) [inline solaris/solaris11/postinstall.sh:6]: Moving log files without asking the user? very sensitive change.

File: solaris/solaris11/smf_manifest.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+            <service_fmri value="svc:/milestone/multi-user-server" />
+        </dependent>
+        <exec_method timeout_seconds="60" type="method" name="start"
+                     exec="var/ossec/installation_scripts/postinstall.sh"/>
+        <property_group type="framework" name="startd">
+                     <propval type="astring" name="duration" value="transient"/>
+        </property_group>
+        <instance enabled="true" name="default"> </instance>
+    </service>
+</service_bundle>
```
Comment (okynos) [inline solaris/solaris11/smf_manifest.xml:21]: Why need we this?

**Review by okynos — CHANGES_REQUESTED** [review]
Review comments, 👍🏼

File: solaris/solaris11/generate_wazuh_packages.sh (hunk 2 above)
Comment (DFolchA) [reply solaris/solaris11/generate_wazuh_packages.sh:173]: In this issue wazuh/wazuh#6885, `var/ossec/logs/ossec `and `var/ossec/queue/ossec` were renamed to `var/ossec/logs/wazuh` and `var/ossec/queue/sockets` respectively, `/var/ossec/logs/wazuh `is ccreated at compilation time.

File: solaris/solaris11/postinstall.sh (hunk 3 above)
Comment (DFolchA) [reply solaris/solaris11/postinstall.sh:6]: After this issue wazuh/wazuh#6885, `var/ossec/logs/ossec` will ont exist anymore this allows us to maintain the old logs.

File: solaris/solaris11/smf_manifest.xml (hunk 4 above)
Comment (DFolchA) [reply solaris/solaris11/smf_manifest.xml:21]: This creates a service atached to the package, that will run the script after installation, with this we can simulate a postinstallation script.

File: solaris/solaris11/SPECS/template_agent_v4.2.0.json (hunk 1 above)
Comment (DFolchA) [reply solaris/solaris11/SPECS/template_agent_v4.2.0.json:682]: Yhis changes follow this issue wazuh/wazuh#6885, for the moment the installation directory will still be `/var/ossec`

**Review by okynos — APPROVED** [review]
LGTM.

**Review by alberpilot — CHANGES_REQUESTED** [review]
Please review

File: solaris/solaris11/SPECS/template_agent_v4.2.0.json [hunk 5]
```diff
@@ -703,287 +1007,287 @@
         "type": "directory",
         "user": "ossec"
     },
-    "/var/ossec/logs/ossec/[0-9][0-9][0-9][0-9]/Jan": {
+      "/var/ossec/logs/ossec/[0-9][0-9][0-9][0-9]/Jan": {
```
[the remark this answers is not in the document]
Comment (DFolchA) [reply solaris/solaris11/SPECS/template_agent_v4.2.0.json]: Done in b0c53d8

**Review by alberpilot — CHANGES_REQUESTED** [review]
5.0 json specs change missing

**Review by alberpilot — APPROVED** [review]

**Pull request merged**